### PR TITLE
Update CODEOWNERS

### PR DESCRIPTION
**Pull request by AlexGhiondea**
/cc @weshaggard @danieljurek @mitchdenny 

I have realized that my tool doesn't yet support labels with spaces. I am going to fix that next (and add more entries to this list).

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -25,30 +25,30 @@
 
 /sdk/digitaltwins/                              @drwill-ms @timtay-microsoft @abhipsaMisra @vinagesh @azabbasi @prmathur-microsoft @bikamani @barustum      %DigitalTwins
 
-/sdk/eventgrid/                                 @Kishp01 @ahamad-MS   
+/sdk/eventgrid/                                 @Kishp01 @ahamad-MS   %Event Grid
```
Comment (weshaggard) [inline .github/CODEOWNERS:28]: As we chatted we might want to encode the labels as a comment. We can certainly try the % but I worry some folks parsing this file might break.

Comment (AlexGhiondea) [reply .github/CODEOWNERS:28]: We talked a bit more about this and the `#` in the same line might also be treated in an undefined way.

We are going to go ahead with the current approach and we will roll it back if that causes problems.

We might consider trying it out in the `azure-sdk-tools` repo first as that has lower churn and lower traffic.

/cc @danieljurek

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -9,46 +9,46 @@
 /sdk/                                           @AlexGhiondea
 
 # Core
-/sdk/core/                                      @pakrym @KrzysztofCwalina
+/sdk/core/                                      @pakrym @KrzysztofCwalina       %Azure.Core
```
Comment (weshaggard) [inline .github/CODEOWNERS:12]: Given we cannot easily align these tags I would just add them all to the end of the users but not a big deal.

**Review by weshaggard — APPROVED** [review]
Looks good to me can you add a note about this in our code owners doc at https://github.com/Azure/azure-sdk/blob/master/docs/policies/opensource.md#codeowners.

File: .github/CODEOWNERS (hunk 2 above)
Comment (AlexGhiondea) [reply .github/CODEOWNERS]: Makes sense. Done.

**Pull request merged**